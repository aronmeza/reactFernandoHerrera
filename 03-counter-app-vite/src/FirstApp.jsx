
const persona1 = {
  message: 'Hola',
  title: 'soy un Subtitulo',
};

const getSaludo = (persona) =>{
  return(
    <>
      <h1>{persona.message}</h1>
      <p>Soy {persona.title}</p>
    </>
  );
}

// export const FirstApp = () => {
//
//   return (
//     <>
//       {/*<code> { JSON.stringify(newMessage) } </code>*/}
//       <h1>{getSaludo(persona1)}</h1>
//     </>);
// }

import PropTypes from "prop-types";
export const FirstApp = ({title, subTitle, name}) => {

  return (
    <>
      <h1 data-testid='test-title'>{title}</h1>
      <p>{subTitle}</p>
      <p>{subTitle}</p>
      <p>{name}</p>
    </>);
}




FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string
}

FirstApp.defaultProps = {
  title : "No titulo",
  subTitle: "Soy un subtitulo",
  name: 'Fernando Herrera'
}