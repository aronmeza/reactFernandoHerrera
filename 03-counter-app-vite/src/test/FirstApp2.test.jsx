import {FirstApp} from "../FirstApp.jsx";
import {render, screen} from "@testing-library/react";

describe('Puebas en  FirstApp.jsx', () => {

  const title = 'Soy Aaron';
  const subTitle = 'Soy subtitulo';

  test('debe de mhacer match con el snapshot', () => {

    const {container, getByText, getByTestId} = render(<FirstApp title={title}/>)
    expect(container).toMatchSnapshot();

  })

  test('debe de contener el mensaje "Soy Aaron', () => {

    render(<FirstApp title={title}/>)
    expect(screen.getByText(title)).toBeTruthy();

  })

  test('debe de mostrar el titulo h1', () => {

    render(<FirstApp title={title}/>)
    expect(screen.getByRole('heading', {level: 1}).innerHTML).toContain(title);

  })

  test('debe de mostrar el subtitulo enviado por props', () => {

    render(<FirstApp title={title} subTitle={subTitle}/>)
    expect(screen.getAllByText(subTitle).length).toBe(2);

  })

});