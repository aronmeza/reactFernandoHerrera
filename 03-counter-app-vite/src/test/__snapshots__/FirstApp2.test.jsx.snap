// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Puebas en  FirstApp.jsx debe de mhacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Soy Aaron
  </h1>
  <p>
    Soy un subtitulo
  </p>
  <p>
    Soy un subtitulo
  </p>
  <p>
    Fernando Herrera
  </p>
</div>
`;
